ir/base.def

```diff
@@ -206,26 +206,41 @@ class AnnotationToken {
 /// Most P4 entities can be optionally annotated
 class Annotation {
     ID name;
+
     /// An unparsed annotation body
     inline Vector<AnnotationToken> body;
+
+    /// Whether the annotation body needs to be parsed.
+    /// Invariant: if this is true, then expr and kv must both be empty. If the
+    /// annotation is compiler-generated (e.g., derived from a P4₁₄ pragma),
+    /// then needsParsing will be false and the body will be empty, but expr or
+    /// kv may be populated.
+    bool needsParsing;
+
     /// Annotations that are simple expressions
     inline Vector<Expression> expr;
+
     /// Annotations described as key-value pairs
     inline IndexedVector<NamedExpression> kv;
+
     Annotation { if (!srcInfo) srcInfo = name.srcInfo; }
+
+    /// For annotations parsed from P4₁₆ source.
```

Comment:
I worry a bit about non-ascii characters in source. The preferred spelling is P4-16 if you want to stick to ASCII.
Makes it also easier to grep.

Reply:
I think we all should embrace the future and be less squeamish about using Unicode. But maybe my years of staring at lattices (viz. ⊔,⊓,⊑) has desensitized me! :smiley:

I buy the grep argument, though. Fixed.